crates/net/src/http/mod.rs

```diff
@@ -119,6 +123,44 @@ impl Request {
         self
     }
 
+    /// Append query parameters to the url, given as `(name, value)` tuples. Values can be of any
+    /// type that implements [`ToString`].
+    ///
+    /// It is possible to append the same parameters with the same name multiple times, so
+    /// `.query([("a", "1"), ("a", "2")])` results in the query string `a=1&a=2`.
+    ///
+    /// # Examples
+    ///
+    /// The query parameters can be passed as many different types:
+    ///
+    /// ```
+    /// # fn no_run() {
+    /// use std::collections::HashMap;
+    /// use gloo_net::http::Request;
+    ///
+    /// let slice_params = [("key", "value")];
+    /// let vec_params = vec![("a", 3), ("b", 4)];
+    /// let mut map_params: HashMap<&'static str, &'static str> = HashMap::new();
+    /// map_params.insert("key", "another_value");
+    ///
+    /// let r = Request::get("/search")
+    ///     .query(slice_params)
+    ///     .query(vec_params)
+    ///     .query(map_params);
+    /// // Result URL: /search?key=value&a=3&b=4&key=another_value
+    /// # }
+    /// ```
+    pub fn query<'a, T, V>(self, params: T) -> Self
+    where
+        T: IntoIterator<Item = (&'a str, V)>,
+        V: ToString,
```

Comment:
Is it to possible to make this `AsRef<str>`? We can avoid the clone that way/

Reply:
I chose `ToString` because it allows to pass other types than `&str`/`String` and I don't see a way to preserve this with `AsRef<str>`. But I agree that the additional clone is not optimal.

Reply:
`AsRef<str>` should also allow that. I believe any type that can be converted to `&str` will fall under that

Reply:
When replacing `ToString` with `AsRef<str>` I get the following error:

```
error[E0277]: the trait bound `{integer}: AsRef<str>` is not satisfied
  --> src/http/mod.rs:150:12
   |
15 |     .query(vec_params)
   |      ----- ^^^^^^^^^^ the trait `AsRef<str>` is not implemented for `{integer}`
   |      |
   |      required by a bound introduced by this call
   ```

Which fails at the docstring example

Reply:
Or am I misunderstanding your suggestion?

Reply:
The error message makes the allocation explicit. Converting a number to string requires creating a String. We should keep that explicit

Reply:
Its `AsRef<str>` now, I've adjusted the doc example.